### PR TITLE
Group operation and read profiling

### PR DESCRIPTION
**Pull request by dtebbs**
- Profiling of group operations and group element reads
- Trivial optimization to avoid perpetually recomputing ncommonly used `one` and `zero` field elements

### CODE REVIEW

File: libff/algebra/curves/profile/algebra_groups_profile.cpp [hunk 1]
```diff
@@ -5,13 +5,54 @@
...
 
 #include <exception>
 
 using namespace libff;
 
+template<typename GroupT>
+bool profile_group_add()
+{
+    static const size_t NUM_DIFFERENT_ELEMENTS = 1024;
+    static const size_t NUM_ELEMENTS = 1024*1024;
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply libff/algebra/curves/profile/algebra_groups_profile.cpp:20]: It's pretty arbitrary.  It takes time to generate random elements and 1024 keeps this time pretty short, but of course we need a reasonably high total number in order to get an accurate measure of cost.  But we could select other numbers here.

[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply libff/algebra/curves/profile/algebra_groups_profile.cpp:20]: I see, thanks

**Review by AntoineRondelet — APPROVED** [review]
LGTM

**Comment by AntoineRondelet** [comment]
>     * Trivial optimization to avoid perpetually recomputing ncommonly used `one` and `zero` field elements

See: #37

**Pull request merged**